transaction/core/src/amount/mod.rs

```diff
@@ -93,30 +71,58 @@ impl MaskedAmount {
     /// Value is denominated in picoMOB.
     ///
     /// # Arguments
-    /// * `shared_secret` - The shared secret, e.g. `rB`.
+    /// * `tx_out_shared_secret` - The shared secret, e.g. `rB`.
     pub fn get_value(
         &self,
-        shared_secret: &RistrettoPublic,
+        tx_out_shared_secret: &RistrettoPublic,
     ) -> Result<(Amount, Scalar), AmountError> {
-        let (expected_commitment, amount, blinding) =
-            Self::compute_commitment(self.masked_value, &self.masked_token_id, shared_secret)?;
-        if self.commitment != expected_commitment {
-            // The commitment does not agree with the provided value and blinding.
-            // This either means that the commitment does not correspond to the shared
-            // secret, or that the amount is malformed (and is probably not
-            // spendable).
-            return Err(AmountError::InconsistentCommitment);
+        match &self {
+            Self::V1(masked_amount) => masked_amount.get_value(tx_out_shared_secret),
+            Self::V2(masked_amount) => masked_amount.get_value(tx_out_shared_secret),
+        }
+    }
+
+    /// Get the masked value field
+    pub fn get_masked_value(&self) -> &u64 {
```

Comment:
[Rust style guide says to omit `get_` prefix](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter) (throughout this PR)